engine/src/main/java/org/pentaho/di/trans/steps/dimensionlookup/DimensionLookup.java

```diff
@@ -0,0 +1,1762 @@
+            columnLookupArray[ i ] = -1;
+          }
+        }
+        // Integer returnRowColNum = null;
+        int returnRowColNum = -1;
+        String findColumn = null;
+        for ( int i = 0; i < meta.getFieldStream().length; i++ ) {
+          if ( data.fieldnrs[ i ] >= 0 ) {
+            // Only compare real fields, not last updated row, last version, etc
+            //
+            ValueMetaInterface v1 = data.outputRowMeta.getValueMeta( data.fieldnrs[ i ] );
+            Object valueData1 = row[ data.fieldnrs[ i ] ];
+            findColumn = meta.getFieldStream()[ i ];
+            // find the returnRowMeta based on the field in the fieldLookup list
+            ValueMetaInterface v2 = null;
+            Object valueData2 = null;
+            // Fix for PDI-8122
+            // See if it's already been computed.
+            returnRowColNum = columnLookupArray[ i ];
+            if ( returnRowColNum == -1 ) {
+              // It hasn't been found yet - search the list and make sure we're comparing
+              // the right column to the right column.
+              for ( int j = 2; j < data.returnRowMeta.size(); j++ ) { // starting at 2 because I know that 0 and 1 are
+                // poked in by Kettle.
+                v2 = data.returnRowMeta.getValueMeta( j );
+                if ( ( v2.getName() != null ) && ( v2.getName().equalsIgnoreCase( findColumn ) ) ) { // is this the
+                  // right column?
+                  columnLookupArray[ i ] =
+                    j; // yes - record the "j" into the columnLookupArray at [i] for the next time
+                  // through the loop
+                  valueData2 = returnRow[ j ]; // get the valueData2 for comparison
+                  break; // get outta here.
+                } else {
+                  // Reset to null because otherwise, we'll get a false finding at the end.
+                  // This could be optimized to use a temporary variable to avoid the repeated set if necessary
+                  // but it will never be as slow as the database lookup anyway
+                  v2 = null;
+                }
+              }
+            } else {
+              // We have a value in the columnLookupArray - use the value stored there.
+              v2 = data.returnRowMeta.getValueMeta( returnRowColNum );
+              valueData2 = returnRow[ returnRowColNum ];
+            }
+            if ( v2 == null ) {
+              // If we made it here, then maybe someone tweaked the XML in the transformation
+              // and we're matching a stream column to a column that doesn't really exist. Throw an exception.
+              throw new KettleStepException( BaseMessages.getString(
+                PKG, "DimensionLookup.Exception.ErrorDetectedInComparingFields", meta.getFieldStream()[ i ] ) );
+            }
+
+            try {
+              cmp = v1.compare( valueData1, v2, valueData2 );
+            } catch ( ClassCastException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)